### ISSUE TITLE
Matting on Mobile Device?

### ISSUE THREAD

**Issue by InternetMaster1** [HymEric/Segmentation-Series-Chaos#1]
Thank you for the amazing library! Wow.... It's an ocean of knowledge.

Could you please recommend a good library to achieve high accuracy segmentation (i.e. matting) on mobile devices?

Much appreciated!

**Comment by HymEric** [comment]
Thank you. You can use this for matting: https://github.com/HymEric/Segmentation-Series-Chaos/tree/master/MMnet-exp.

**Comment by InternetMaster1** [comment]
Wow, thank you for your quick response. I will most definitely try out your recommendation.

I had a quick question. I am assuming MMTNet is an end-to-end solution without the need for a manual trimap.

How would the accuracy of MMTNet compare to other matting solutions which
require a trimap (say IndexNet Matting)? Have you had a chance to evaluate
lightweight trimap based matting solutions catering to mobile devices? And in such cases, generating trimap automatically with opencv solutions (such as erode and dilate) would give a good quality trimap? 

Thanks!

**Comment by InternetMaster1** [comment]
I was trying to locate the MMTNet algorithm ranking on the alphamatting.com website but am unable to locate it. Can you help me out?

**Comment by HymEric** [comment]
What's the MMTNet? Or the MMNet? I think trimap is the tool to improve the accuracy of matting. However, if you want to end-to-end matting method without trimap, it's ok. Sorry, I didn't use the trimap based method. Maybe you can try it.

**Comment by InternetMaster1** [comment]
Sorry, I meant MMNet.

OK thanks a lot for your feedback, I truly appreciate it.

**Comment by InternetMaster1** [comment]
Dear Yongming He,

I am having a little trouble running the MMNet library. 

I am using https://github.com/hyperconnect/MMNet repository and wanted to use your MMnet-exp training checkpoints.

I am a little lost about how to run inference and I did not find any file to run trained checkpoints for getting cutout.

Could you guide me?

Thanks a ton!

**Comment by HymEric** [comment]
Sorry for replying late. You can check *./scripts/valid_mmnet_dm1.0_256.sh* and *evaluate.py* in  https://github.com/hyperconnect/MMNet. Note that you should modify codes based on your project.

**Comment by InternetMaster1** [comment]
Dear Yongming He,

I tried your suggestions. 

I created two directories, 

1) First one is "image_dataset" which contains "test/image" directory for images and "test/mask" directory for mask
2) Second one is "models" in which I have copied your 
https://github.com/HymEric/Segmentation-Series-Chaos/tree/master/MMnet-exp/1898000-exp model files

Then I ran the following command

> !sh scripts/valid_mmnet_dm1.0_256.sh "/image_dataset" "/models"

But execution is stuck at below for loop line. 65 in MMNet/common/tf.utils.py (https://github.com/hyperconnect/MMNet/blob/master/common/tf_utils.py)
and not moving forward and even not giving any error

from tensorflow.contrib.training import checkpoints_iterator
for ckpt_path in checkpoints_iterator(checkpoint_dir, min_interval_secs, timeout, timeout_fn):

Do you have any rough idea about what exactly I have missed. Once again, sorry for the newbie questions.

**Comment by HymEric** [comment]
It's been a long time. Sorry, I don't remember it clearly. If I have time, I would check it and give the solution. I modified some codes based on the MMNet for my own training and testing in the project. You should first read the codes carefully and then modify it according your project, to improve it. If you encounter some specific problems,  you better ask the author of MMNet repo. Notes: maybe my trained checkpoints don't suit for original MMNet model for I remember I've changed the BN related layer or others.

**Comment by HymEric** [comment]
And you can refer my another repo about matting for a quick test: https://github.com/HymEric/AIWebMatting

**Comment by InternetMaster1** [comment]
Thanks for the response.

I will check out the new library that you gave me.

Do you think you could have the code for your modified MMNet library? These
things are complex and I am having a tough time making it run. Any help you
could provide would be greatly appreciated.

**Comment by InternetMaster1** [comment]
> And you can refer my another repo about matting for a quick test: https://github.com/HymEric/AIWebMatting

In this demo on http://matting.zsyhh.com:4800/, have you used MMNet algorithm along with you pre-trained model  found at your[ MMNet-exp library](https://github.com/HymEric/Segmentation-Series-Chaos/tree/master/MMnet-exp/1898000-exp)?

**Comment by HymEric** [comment]
Exactly. One of my pre-trained model.

**Comment by InternetMaster1** [comment]
Thanks :)
